src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -0,0 +1,1345 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Runtime.CompilerServices;
+using System.Globalization;
+using System.Runtime.Versioning;
+using System.Diagnostics.Contracts;
+
+using Windows.Storage.Streams;
+using Windows.Security.Cryptography;
+
+namespace System.IO
+{
+    // Provides methods for processing directory strings in an ideally
+    // cross-platform manner.  Most of the methods don't do a complete
+    // full parsing (such as examining a UNC hostname), but they will
+    // handle most string operations.  
+    // 
+    // File names cannot contain backslash (\), slash (/), colon (:),
+    // asterick (*), question mark (?), quote ("), less than (<;), 
+    // greater than (>;), or pipe (|).  The first three are used as directory
+    // separators on various platforms.  Asterick and question mark are treated
+    // as wild cards.  Less than, Greater than, and pipe all redirect input
+    // or output from a program to a file or some combination thereof.  Quotes
+    // are special.
+    // 
+    // We are guaranteeing that Path.SeparatorChar is the correct 
+    // directory separator on all platforms, and we will support 
+    // Path.AltSeparatorChar as well.  To write cross platform
+    // code with minimal pain, you can use slash (/) as a directory separator in
+    // your strings.
+    // Class contains only static data, no need to serialize
+    [ComVisible(true)]
+    public static class Path
+    {
+        // Platform specific directory separator character.  This is backslash
+        // ('\') on Windows, slash ('/') on Unix, and colon (':') on Mac.
+        // 
+#if !PLATFORM_UNIX        
```

Comment:
Another case where the file isn't included in the csproj, so I'm not going to fix this stuff up as part of the PR.  @mellinoe Can help us understand what the right long term thing is.